src/nnet3/nnet-normalize-component.cc

```diff
@@ -221,28 +225,38 @@ void BatchNormComponent::ComputeDerived() {
     stats_sum_.SetRandn();
     stats_sumsq_.SetRandn();
     stats_sumsq_.AddVecVec(1.0, stats_sum_, stats_sum_, 1.0);
+    count_eq_zero = true;
+    // because the moving_mean_ and moving_stddv_ will not be set to zero
```

Comment:
Right now you are not doing anything in the Add()/Scale() functions w.r.t. moving_mean_ and moving_stddev_, so it wouldn't correctly take the average when averaging models, it would just take one of them.
Normally it's better to represent and average these things as stats, not things like mean and stddev, but it's not how the guys wrote the paper.  (That would be easier w.r.t. the Scale() function).

Reply:
Yes, at this stage, I just pick one of the moving_mean/var of the parallel models (normally it should be the nnet[0] ). 
It's hard to do the sum and scale which we used for batchnorm for batch-renorm.
I think one way to improve is average the moving_mean/var of the parallel jobs.

Reply:
But my experiments showed that, this way (pick moving_mean/var of one of the parallel jobs but did nothing like averaging or so) worked well. I guess the averaging is minor effect for batch-renorm.